rainier-cats/src/main/scala/com/stripe/rainier/cats/package.scala

```diff
@@ -83,3 +126,39 @@ private[cats] class MonadRandomVariable extends Monad[RandomVariable] {
       case Right(b) => RandomVariable(b)
     }
 }
+
+private[cats] trait EqInstances {
+
+  def eqBigDecimal(epsilon: Double): Eq[BigDecimal] =
+    Eq.instance { (left, right) =>
+      if (right == BigDecimal(0.0))
+        left.abs < epsilon
+      else ((left - right).abs / right.abs) < epsilon
+    }
+
+  def eqReal(epsilon: Double): Eq[Real] = {
+    val bde = eqBigDecimal(epsilon)
+    Eq.instance { (left, right) =>
+      (left, right) match {
+        case (Infinity, Infinity) | (NegInfinity, NegInfinity) => true
+        case (Constant(a), Constant(b))                        => bde.eqv(a, b)
+        case _                                                 => false
+      }
+    }
+  }
+
+  implicit val defaultEqReal: Eq[Real] = eqReal(1e-6)
+
+  implicit def eqCategorical[A: Eq]: Eq[Categorical[A]] = {
+    implicit val mapEq: Eq[Map[A, Real]] = catsKernelStdEqForMap[A, Real]
+    Eq.by(_.pmf)
+  }
+
+  implicit def eqGenerator[A](
+      implicit eqA: Eq[A],
+      r: RNG,
+      n: Numeric[Real]
+  ): Eq[Generator[A]] = Eq.by(_.get)
+
+  implicit def eqRandomVariable[A: Eq]: Eq[RandomVariable[A]] = Eq.by(_.value)
```

Comment:
This makes me cringe a bit - it's fine if it narrowly serves your needs but two RVs with equal values and different densities are definitely not equal. I'd suggest at least warning about that in comments.

Reply:
Warning added - my mistake, I don't get how to write a proper `Eq` method that compares densities yet. Happy to do so with some guidance, perhaps in the next PR?